### PR TITLE
F/implement subgit import

### LINKED ISSUE #34: Implement gitlab/github import feature
So @naestia had the idea that we should be able to generate or "import" all git repos that is available from either my user account or an organisation account.

We have an example script of this already made on the side that supports this for our own needs, but this should be supported in the main tool if possible.

Implement the cli commands similar to this

```
subgit import github <username-or-organisation> <output-file>
subgit import gitlab <username-or-organisation-id> <output-file>
```

where we also define the output file. Note here that we need protection to avoid or to ask the user to overwrite existing configuration file. The output file maybe should be optional and it should default to `.subgit.yml` by default.

Things we should parse out from the repo is the clone url, repo name and default branch name and save this in the configuration file. Anything else is redundant by default and needs to be added by the user manually after.

So one thing here and one critical thing is that any dependencies we have or need in order to make the commands work like git-python or if we need any other libraries based on what the script translation will be, if we need additional python dependencies, they should be put in the extras_installs section in setup.py file and the commands should be enabled/disabled based on if the python imports exists and can be done.

### PR DESCRIPTION
**Pull request by naestia**
Fixed #34 

Implemented 'subgit import' with access to github and gitlab.

Fixed bugs related to subgit import which affected other commands. One of which being 'subgit pull' not being able to pull from config file where branch name is empty.

Updated README to include how to get started with 'subgit import' since it requires aditional dependencies which subgit does not help the user aquire.

### CODE REVIEW

File: subgit/importer/git_importer.py
Comment (Grokzen) [inline subgit/importer/git_importer.py:73]:
```suggestion
        out = subprocess.run([
                "gh", "repo", "list",
                f"{owner}",
                "--json", "id,name,defaultBranchRef,sshUrl,isArchived",
                "-L", "100"
            ],
```

So this suggestion is very unconventional as it would put multiple strings in a list on the same line, but at the same time it would increase the readability of the code as we join together what arguments really belong to eachother much more clearly.

You can choose here to keep as is or take this suggestion. by taking it we break pep8 a bit tho :P

**Review by Grokzen — CHANGES_REQUESTED** [review]
Some comments and some required changes needed. But otherwise looks good. Remember to also check the sub group problem in gitlab first.

If you push additional changes or fixes for these, remember to request a second review once you are done with all things before we merge later

File: subgit/importer/git_importer.py
Comment (naestia) [reply subgit/importer/git_importer.py:73]: Given that it's structured in a similar way in the other method with multiple strings in the same row, I'd say you have a good point :P

File: subgit/importer/git_importer.py
Comment (naestia) [reply subgit/importer/git_importer.py:36]: Yeah, good catch.. It was my intention to change that, but totally forgot about that :P

**Comment by naestia** [comment]
I also tested using subgit import with the sub-group and it worked like a charm

**Review by Grokzen — APPROVED** [review]
LGTM, lets run with this

**Pull request merged**
